### DISCUSSION TITLE
New H4 installation on my JETBD920

### DISCUSSION THREAD

**Discussion by Kisssys** [kachurovskiy/nanoels#121] [Show and tell]
I have been following Maxim for a while and was amazed what he had done with the Nano but that just didn't get me onboard until
he created the H4 with the ESP32.

I ordered the H4 when I saw how easy it was going to be to mount the X axis servo.  Unfortunatly although or lathes are simular the 
distance from the center of the X axis lead and the mounting screws was off a couple of mm's.  I will some day draw a new mount
but I decided to just weld up a mount that uses som existing screws ( just need to use longer screws );  

I had a Nema23 motor so I drew up gears that would make it all fit in the limited space. I just uploaded the files for my 2 axis Hand wheel, my encoder mount and my X and Z axis mounts to Thingiverse and Printable's under Kisssys.

I had a few bumps but it came together pretty quickly.  Maxim has done a good job to protect the H4 but using a cheap encoder caused me to make a few changes to the H4 which I wouldn't recommend if you don't know what you're doing.

I'm still learning to use the interface, I don't have the setting of the limits down pat.  I think sometimes I have a limit set and then unset it when I think I'm setting it.  Need to just use it for a few month's.

These are links to all the changes to my JetBD920N and my 2 axis Handwheel which I am really liking.

https://www.printables.com/model/498325-jetbd920-servo-mounts
https://www.printables.com/model/498324-jetbd920-encoder-mount
https://www.printables.com/model/497597-handwheel-pulse-encoder-dual-axis

I did a drawing of my x axis cross slide a long time ago and posted a link to the drawing with some pictures that might be of use
to someone with the BD920.

https://www.thingiverse.com/thing:3272285

I can't imagine that someone won't build a low cost VFD controlled lathe with good quality ballscrews on Z and X and leave out all the junk you don't need anymore.

Hope this was helpful.
Cheers
Kisssys
![40to20gearNema23Zmount](https://github.com/kachurovskiy/nanoels/assets/5122480/5e176a90-2294-47f1-89d2-01d8bd90fc02)
![EncoderMount](https://github.com/kachurovskiy/nanoels/assets/5122480/1c451d70-6bf9-43ff-ad87-0831b2ffc1c8)
![EncoderMount12mmShaftZextension](https://github.com/kachurovskiy/nanoels/assets/5122480/72443259-a48d-47d4-9644-6ccc146bd295)
![EncoderMountJetBD920](https://github.com/kachurovskiy/nanoels/assets/5122480/777096e3-5dc2-4da5-88ae-0b5290e0460a)
![IMG_1174](https://github.com/kachurovskiy/nanoels/assets/5122480/e4ebd625-5e7f-47aa-8a66-a610f2ffb5aa)
![IMG_1179](https://github.com/kachurovskiy/nanoels/assets/5122480/ee09bf27-2105-4827-a67a-132243fe5ca8)
![IMG_1180](https://github.com/kachurovskiy/nanoels/assets/5122480/50410770-a2a5-4451-a190-276f2731bf5b)
![JetBD920zAxisNema23moutn](https://github.com/kachurovskiy/nanoels/assets/5122480/db22fd85-cc1b-42a4-a332-d1682ff2bed0)
![Nema23BD920](https://github.com/kachurovskiy/nanoels/assets/5122480/87fc317f-69de-4900-8916-6fc0e907f210)
![XaxisNema23mount](https://github.com/kachurovskiy/nanoels/assets/5122480/f383f4cc-e834-46a3-943f-e30029668b91)

**Comment by kachurovskiy** [comment]
Hey @Kisssys, thanks a lot for sharing your build. Sorry to hear that limits are giving you trouble, is there something we can do to make them move self-explanatory? Perhaps change `LRUD` to icons `←→↑↓`? I also thought about showing the distance to the upcoming limit in the bottom display line.

**Reply by Kisssys** [reply]
Well I just played with it for a while and I'm a dummy, that is fine just the way it is.

Cheers

**Reply by Kisssys** [reply]
I just did the latest update and I like the symbols for limits better than LRUD.  It would sure be nice to be able to press a button 
and see what the limits are.  I have a tendancy to run to limits to make sure I've set them correctly which is time consuming
and sometimes not convenient to do.  

If I ask the Z axis to move to fast when turning or cutting threads my closed loop servo faults.   This is my fault and I know I've asked too
much of it but is their a setting I can make that would limit my ability to set a unreasonable movement. 

Cheers
Kisssys

**Reply by kachurovskiy** [reply]
> It would sure be nice to be able to press a button and see what the limits are

Would it help if bottom line would show whether you're on a limit and the distance to the upcoming limit? E.g. `At ←, 3.55" to →`?

> If I ask the Z axis to move to fast when turning or cutting threads my closed loop servo faults

Does it happen during the early acceleration phase, when running at constant speed or when stopping? Try reducing Z axis acceleration in https://github.com/kachurovskiy/nanoels/blob/main/h4/h4.ino#L17 - also, seeing a video of the event would help understand what could be the problem.

Thank you!

**Comment by Kisssys** [comment]
> Would it help if bottom line would show whether you're on a limit and the distance to the upcoming limit? E.g. At ←, 3.55" to →?

Yes I think that would work, be worth trying it out a while.  I need to know that pressing go will be within reason. I made the change you
suggested in removing all three lines of code for the position display and I was happy with that.  I don't think showing "Position 0"
makes sense.

> Does it happen during the early acceleration phase, when running at constant speed or when stopping? Try reducing Z axis acceleration in https://github.com/kachurovskiy/nanoels/blob/main/h4/h4.ino#L17 - also, seeing a video of the event would help understand what could be the problem.

I would say it happens during the early acceleration stages,  When I get a chance I'll do a little more trouble shooting and experiment with Z accel. I need to hook a scope  to Z step, maybe my logic analyzer would be better. I've got a lot of other projects going on but when I get a chance I'll play with it.

Also, on 2 occasions during operation I got the "key down at power up hardware failure"  I am going to guess I had 2 buttons pressed at the same time but I thought I'd mention it since it shuts down until you turn power on and off.

I'm very pleased overall and have done a number of simple parts already and I have not once opened the half nut so that says I'm fairly comfortable with how it operates as I don't revert back to total manual operation.  Part of the success is the hand encoder on Z and X which I use a lot.

Cheers
Steve

**Reply by kachurovskiy** [reply]
> I would say it happens during the early acceleration stages

Reducing `ACCELERATION_Z` should definitely help here, https://github.com/kachurovskiy/nanoels/blob/main/h4/h4.ino#L17

> Also, on 2 occasions during operation I got the "key down at power up hardware failure"

This could have happened before but should be fixed in the latest version.

Thanks for the feedback!

**Comment by Kisssys** [comment]
Thanks
